nautobot/core/utils/lookup.py

```diff
@@ -218,3 +220,122 @@ def get_created_and_last_updated_usernames_for_model(instance):
         last_updated_by = last_updated_by_record.user_name
 
     return created_by, last_updated_by
+
+
+def get_url_patterns(urlconf=None, patterns_list=None, base_path="/"):
+    """
+    Recursively yield a list of registered URL patterns.
+
+    Args:
+        urlconf (URLConf): Python module such as `nautobot.core.urls`.
+        patterns_list (list): Used in recursion
+        base_path (str): String to prepend to all URL patterns yielded.
```

Comment:
Add default values for these

Reply:
Can we add the derived defaults here? I'm thinking the default of `settings.ROOT_URLCONF` if the `urlconf` kwarg is `None`

Reply:
Ah sure, good thought.